WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -91,15 +91,14 @@ def destroy
     end
   end
 
-  private def selected_registrations_from_params
-    params[:selected_registrations].map { |r| r.split('-')[1] }.map do |registration_id|
-      competition_from_params.registrations.find_by_id!(registration_id)
-    end
+  private def selected_registrations_from_params(competition_registrations)
+    registration_ids = params[:selected_registrations].map { |r| r.split('-')[1] }
+    competition_registrations.find(registration_ids)
   end
 
   def export
     @competition = competition_from_params
-    @registrations = selected_registrations_from_params
+    @registrations = selected_registrations_from_params(@competition.registrations.includes(:user, :events))
```

Comment:
Whatever you do end up doing here, probably best to share the `includes` logic with the `do_actions_for_selected` method.